sos/archive.py

```diff
@@ -209,7 +210,14 @@ def add_node(self, path, mode, device):
         dest = self.dest_path(path)
         self._check_path(dest)
         if not os.path.exists(dest):
-            os.mknod(dest, mode, device)
+            try:
+                os.mknod(dest, mode, device)
+            except OSError as e:
```

Comment:
There's no need for that here, really: sos is just taking an archive of a directory - in the case that we can create the node, it carries a little extra, useful information with it. It's not enough use to justify using our own private namespace (and in any case, that will not work if the container process has dropped a necessary capability - iirc CAP_SYS_ADMIN, or CAP_RAW_IO for mknod()).

We'll collect the node opportunistically as Louis' patch does.